### PR TITLE
More Spark integration tests and plan visitors

### PR DESCRIPTION
**Pull request by collado-mike**
Added test cases for writing to JDBC and also INSERT OVERWRITE DIRECTORY queries. Discovered a lot of gaps in plan visiting, so I added the necessary handlers.

### CODE REVIEW

File: integrations/spark/build.gradle [hunk 1]
```diff
@@ -48,6 +48,10 @@ dependencies {
     testImplementation platform('org.junit:junit-bom:5.7.1')
     testImplementation "org.postgresql:postgresql:${postgresqlVersion}"
     testImplementation 'org.hamcrest:hamcrest-library:2.2'
+    testImplementation('org.xerial:sqlite-jdbc:3.34.0')
+    testCompile 'org.projectlombok:lombok:1.18.18'
```
Comment (wslulciuc) [inline integrations/spark/build.gradle:52]: Lombok should already be set as a dependency via the root [`build.gradle`](https://github.com/MarquezProject/marquez/blob/main/build.gradle#L76).

File: integrations/spark/integrations/sparksql/1.json [hunk 2]
```diff
@@ -58,6 +59,12 @@
             ],
             "child": 0
           },
+          {
+            "class": "org.apache.spark.sql.catalyst.plans.logical.Project",
+            "num-children": 1,
```
Comment (wslulciuc) [inline integrations/spark/integrations/sparksql/1.json:64]: Is there a standardized way we want to name fields or we don't naming control for logical plans?

File: integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/CommandPlanVisitor.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import org.apache.spark.sql.execution.command.InsertIntoDataSourceDirCommand;
+import org.apache.spark.sql.execution.datasources.InsertIntoDataSourceCommand;
+import org.apache.spark.sql.execution.datasources.SaveIntoDataSourceCommand;
+import scala.PartialFunction;
+import scala.runtime.AbstractPartialFunction;
+
+/**
+ * {@link LogicalPlan} visitor that extracts the input query of certain write commands that don't
+ * expose their input {@link LogicalPlan#children()}. Plans that expose their children are traversed
+ * normally by calling {@link LogicalPlan#collect(PartialFunction)}, but children that arnen't
```
Comment (wslulciuc) [inline integrations/spark/src/main/java/marquez/spark/agent/lifecycle/plan/CommandPlanVisitor.java:21]: Typo: _aren't_

File: integrations/spark/src/test/java/marquez/spark/agent/lifecycle/SparkReadWriteIntegTest.java [hunk 4]
```diff
@@ -153,4 +162,125 @@ public void configure(Binder binder) {}
...
+            tableName,
+            new Properties());
+
+    // wait for event processing to complete
+    StaticExecutionContextFactory.waitForExecutionEnd();
+
+    ArgumentCaptor<LineageEvent> lineageEvent = ArgumentCaptor.forClass(LineageEvent.class);
+
+    // FIXME- the DataFrame -> RDD conversion in the JDBCRelationProvider causes two different sets
+    // of job execution events. Both end up triggering the open lineage event creation
```
Comment (wslulciuc) [inline integrations/spark/src/test/java/marquez/spark/agent/lifecycle/SparkReadWriteIntegTest.java:196]: Mind linking the open issue #1197 in the comments?

File: integrations/spark/src/test/java/marquez/spark/agent/lifecycle/SparkReadWriteIntegTest.java [hunk 5]
```diff
@@ -153,4 +162,125 @@ public void configure(Binder binder) {}
...
+        instanceOf(OutputStatisticsFacet.class));
+
+    // SaveIntoDataSourceCommand doesn't accurately report stats :(
+    assertThat(
+        ((OutputStatisticsFacet) output.getFacets().getAdditionalFacets().get("stats"))
+            .getRowCount(),
+        equalTo(0L));
+  }
+
+  private Path writeTestDataToFile(Path writeDir) throws IOException {
```
Comment (wslulciuc) [inline integrations/spark/src/test/java/marquez/spark/agent/lifecycle/SparkReadWriteIntegTest.java:229]: _Minor_: Should we also cleanup the test file?

**Review by wslulciuc — APPROVED** [review]
Some minor comments to address, otherwise 🚢 it

File: integrations/spark/integrations/sparksql/1.json (hunk 2 above)
Comment (collado-mike) [reply integrations/spark/integrations/sparksql/1.json:64]: The logical plan is generated by Spark directly- we just call their `toJson` method.

**Pull request merged**